Eastern-Research-Group/expert-query
# Feature/143 profile slug

## Related Issues:
* [EQ-143](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=865&view=detail&selectedIssue=EQ-143)

## Main Changes:
* Moved the `dataProfile` parameter out of the URL query string and into a route parameter.
* Moved profile-related content (filter fields, download capabilities) into a child route.
* Extracted some hooks for readability. Some may be overkill, but I figured it's a good idea to keep one level of abstraction in the `Home` component.
* Added a conditional to only parse URL query parameters once. This way, URL input is only checked against the first profile and loaded into state. This avoids the need to separate profile state.

## Steps To Test:
1. Navigate to the app's homepage: http://localhost:3000/attains.
2. Select a profile, and confirm the URL's path updates to include the profile name.
3. Build a search query using the input fields, then select a different profile and confirm the selections are maintained.
4. Verify that the copy boxes are updated to the correct URL, and that they still function as expected.
